pincer/commands/commands.py

```diff
@@ -242,16 +255,27 @@ async def test_command(
 
             # TODO: Write better exception
             raise InvalidArgumentAnnotation(
-                "Type must be CommandArg or other valid type"
+                "Type must be Annotated or other valid type"
             )
 
-        command_type = _options_type_link[annotation.command_type]
+        command_type = _options_type_link[argument_type]
+
+        def get_arg(t: T) -> T:
```

Comment:
Since this also returns `MISSING`

```suggestion
        def get_arg(t: T) -> APINullable[T]:
```